File: app/src/main/java/com/stedi/randomimagegenerator/app/di/modules/ActivityModule.java
package com.stedi.randomimagegenerator.app.di.modules;

import android.content.Context;
import android.support.annotation.NonNull;

import com.stedi.randomimagegenerator.app.di.ActivityContext;
import com.stedi.randomimagegenerator.app.view.activity.base.BaseActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
    private final BaseActivity activity;

    public ActivityModule(@NonNull BaseActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityContext
    Context provideActivityContext() {
        return activity;
    }
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/di/modules/ActivityModule.kt
package com.stedi.randomimagegenerator.app.di.modules

import android.content.Context
import com.stedi.randomimagegenerator.app.di.ActivityContext
import com.stedi.randomimagegenerator.app.presenter.impl.*
import com.stedi.randomimagegenerator.app.presenter.interfaces.*
import com.stedi.randomimagegenerator.app.view.activity.base.BaseActivity
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module(includes = [(ActivityModule.Declarations::class)])
class ActivityModule(private val activity: BaseActivity) {
    @Module
    interface Declarations {
        @Binds
        fun provideEditColoredCirclesPresenter(presenter: SimpleIntegerParamsPresenterImpl): SimpleIntegerParamsPresenter

        @Binds
        fun provideColoredNoiseParamsPresenter(presenter: ColoredNoiseParamsPresenterImpl): ColoredNoiseParamsPresenter

        @Binds
        fun provideGenerationPresenter(presenter: GenerationPresenterImpl): GenerationPresenter

        @Binds
        fun providePreviewGenerationPresenter(presenter: PreviewGenerationPresenterImpl): PreviewGenerationPresenter

        @Binds
        fun provideHomePresenter(presenter: HomePresenterImpl): HomePresenter
    }

    @Provides
    @ActivityContext
    fun provideActivityContext(): Context = activity
}